### ISSUE TITLE
Modifying a loaded unet model 

### ISSUE THREAD

**Issue by junsukha** [facebookresearch/ViewDiff#14]
Hi!

This question is not specifically about your model but I would appreciate if you could suggest some advice.

I want to add a custom layer to zero123's unet module. I think it would be convenient if I could use diffusers library if there's any for this.

I see that you load a unet model by using from_pretrained.
Suppose I did load a unet model. How can I add a custom layer to this loaded unet model? Do I have to write a function that does this or is there an easier way like by using diffuers lib functions?

Basically, I want to make a custom unet model based off of a loaded unet model using from_pretrained.

I think you did the similar work here:
https://github.com/facebookresearch/ViewDiff/blob/7df36e54728f760a119dac91a1fad01fab0a71aa/viewdiff/model/custom_unet_2d_condition.py#L809

You first loaded a unet model by using `pre_trained` and converted it to your custom model `UNet2DConditionCrossFrameInExistingAttnModel`. 

Does your method also apply to my task as well?

Thanks!

**Comment by lukasHoel** [comment]
Hi, yes you can do it similarly, however it might not be the fastest option. In retrospect I had rather used the LDM / CompVis codebase to modify the unet. Diffusers is not so great at model customization, i.e. I had to copy-paste a lot of files and modify them. I suggest to judge for yourself the necessary overhead in our models subfolder
